skbuild/cmaker.py

```diff
@@ -66,9 +66,19 @@ def configure(self, clargs=(), generator_id=None):
             The string representing the CMake generator to use.  If None, uses defaults for your platform.
         """
 
-        clargs, generator_id = pop_arg('-G', clargs)
-        generator_id = get_platform().get_best_generator(
-            generator_id)
+        # if no provided default generator_id, check environment
+        if generator_id is None:
+            generator_id = os.environ.get("CMAKE_GENERATOR")
```

Comment:
Good for now. We may have to revisit later based on best practice associated distutils and next generation of python build system discussed on the "distutils" mailing list.